fuel-merkle/src/binary/root_calculator.rs

```diff
@@ -32,6 +32,20 @@ impl MerkleRootCalculator {
         Self { stack }
     }
 
+    pub fn new_from_existing_leafs(leaf_hashes: Vec<Bytes32>) -> Self {
```

Comment:
perhaps `Result<Self>` is better here since we have some error paths in the function

Reply:
unless its qed, if so please mark them as qed